### ISSUE TITLE
NA textures & metadata / factor levels for horizon data from fetchNASISWebReport()

### ISSUE THREAD

**Issue by brownag** [ncss-tech/soilDB#110]
Incomplete choice lists during factor conversion result in large portions of records converted to NA.

Note in the example below that if `stringsAsFactors = TRUE` a very large number of the horizon records will have `<NA>` texture -- whereas they should have a value populated.

```
nombre <- "ES - MLRA 18 - Thermic Low Rolling Hills 14 - 20 PZ R018XI163CA"
res0 <- get_chorizon_from_NASISWebReport(projectname=nombre, fill=FALSE, stringsAsFactors = TRUE)
res1 <- get_chorizon_from_NASISWebReport(projectname=nombre, fill=FALSE, stringsAsFactors = FALSE)
table(res0$texture)
table(res1$texture)
```

This problem boils down to the metadata choice list used when the string->factor conversion takes place.

Load the metadata:
```
metadata <- NULL
load(system.file("data/metadata.rda", package = "soilDB")[1])
```

The choicelist options used by `get_chorizon_from_NASISWebReport` _currently_ are:

```
> metadata[metadata$ColumnPhysicalName == "texcl", "ChoiceName"]
 [1] "cos"  "s"    "fs"   "vfs"  "lcos" "ls"   "lfs"  "lvfs" "cosl" "sl"   "fsl"  "vfsl" "l"    "sil"  "si"   "scl"  "cl"   "sicl" "sc"  "sic" "c" 
```

Whereas in-lieu textures, texture modifiers hyphenated on to class, etc. (e.g. `gr-sil` or `br`) are all fair game for the values that are in result fetched from NASIS web report. In the case where these are the RV horizon texture and they are not in the choice list, NA is returned.

We either need to select a more complete metadata choice list (I am not sure that one exists) or have factor levels calculated from the resulting set (i.e. dependent on what the contents of the set are, rather than all possibilities) 

Notably, if default `stringsAsFactors` option is set to `TRUE`, this function will "lose" data essentially silently.

**Comment by dylanbeaudette** [comment]
Thanks for reporting this. Does anyone know how and how often the `metadata` object is created? I took a look in the 'misc/' directory and didn't find anything.

I am pretty sure that `uncode()` uses field names to perform the lookup, could it be as simple as careful selection of field names in queries to various data sources?

**Comment by smroecker** [comment]
Ok, I fixed this by adding the texcl column, and leaving the texture column as a character (as it should be). However texcl will still return NA where their are 2 RV textures (e.g. sr- s grx-cos). This should be obvious when the texture and texcl columns are compared.

**Comment by brownag** [comment]
Thanks @smroecker -- the fix works. I think this issue can be closed with 08ba798
 -- there a number of other avenues where we are discussing the ways to deal with metadata/uncoding. 

```
devtools::install_github("ncss-tech/soilDB", dependencies = F)
library(soilDB)
nombre <- "ES - MLRA 18 - Thermic Low Rolling Hills 14 - 20 PZ R018XI163CA"
res0 <- get_chorizon_from_NASISWebReport(projectname=nombre, fill=FALSE, stringsAsFactors = TRUE)
res1 <- get_chorizon_from_NASISWebReport(projectname=nombre, fill=FALSE, stringsAsFactors = FALSE)
testthat::expect_equivalent(table(res0$texture),table(res1$texture))
```
